pkg/pidhandle/pidhandle_linux.go

```diff
@@ -0,0 +1,186 @@
+//go:build linux
+
+// Package for handling processes and PIDs.
+package pidhandle
+
+import (
+	"encoding/hex"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+	"golang.org/x/sys/unix"
+)
+
+type pidfdHandle struct {
+	pidfd        int
+	normalHandle pidHandle
+}
+
+// Store the "unix." methods in variables so we can mock them
+// in the unit-tests and test out different return value.
+var (
+	pidfdOpen       = unix.PidfdOpen
+	newFileHandle   = unix.NewFileHandle
+	openByHandleAt  = unix.OpenByHandleAt
+	nameToHandleAt  = unix.NameToHandleAt
+	pidfdSendSignal = unix.PidfdSendSignal
+)
+
+// Creates new PIDHandle for a given process pid.
+func NewPIDHandle(pid int) (PIDHandle, error) {
```

Comment:
This needs a comment that this can be racy if the pid is already reaped, it is the caller responsibility to ensure that either cannot happen or accept the risk of the pid resuse race.

Long term we likely also want a NewPIDHandleFromPidfd or something if we add callers that already have a pidfd but since that is not the case right now we can add that later